base_content_space.sol

```diff
@@ -406,39 +494,56 @@ contract BaseContentFactory is Ownable {
+        bytes memory cObjBin = ContentFactoryHelper(factoryHelper).getContentObjectBin();
+        address payable contentAddr = createContractBytes(cObjBin, msg.sender, lib, content_type);
+        BaseContent content = BaseContent(contentAddr);
+
         content.setAddressKMS(libraryObj.addressKMS());
         content.setContentContractAddress(libraryObj.contentTypeContracts(content_type));
 
-        // register object in user wallet
-        BaseContent(content).setRights(tx.origin, 0 /*TYPE_SEE*/, 2 /*ACCESS_CONFIRMED*/);
-        content.transferOwnership(tx.origin);
+        IUserSpace userSpaceObj = IUserSpace(contentSpace);
+        address payable userWallet = address(uint160(userSpaceObj.userWallets(tx.origin)));
+
+        // due to a (intentional?) limitation of the EVM, this stanza *needs* to be duplicated as-is. in particular,
+        //  factoring this code into a shared subroutine will cause it to fail.
+        bool isV3Contract = Utils.checkV3Contract(userWallet);
+        if (!isV3Contract) {
+            AccessIndexor index = AccessIndexor(userWallet);
+            content.transferOwnership(tx.origin);
+            index.setContentObjectRights(address(content), 0, 2);
+        } else {
+            // v3+ path ...
+            BaseContent(content).setRights(tx.origin, 0 /*TYPE_SEE*/, 2 /*ACCESS_CONFIRMED*/);
+            content.transferOwnership(tx.origin);
+        }
 
         return address(content);
     }
 
     uint32 public constant OP_ACCESS_REQUEST = 1;
     uint32 public constant OP_ACCESS_COMPLETE = 2;
 
-    function isContract(address addr) returns (bool) {
+    function isContract(address addr) public view returns (bool) {
         uint size;
         assembly { size := extcodesize(addr) }
         return size > 0;
     }
 
-    function executeAccessBatch(uint32[] _opCodes, address[] _contentAddrs, address[] _userAddrs, uint256[] _requestNonces, bytes32[] _ctxHashes, uint256[] _ts, uint256[] _amt) public {
+    function executeAccessBatch(
+        uint32[] memory _opCodes,
+        address payable[] memory _contentAddrs,
+        address[] memory _userAddrs,
+        uint256[] memory _requestNonces,
+        bytes32[] memory _ctxHashes,
+        uint256[] memory _ts,
+        uint256[] memory) public {
 
-        //        BaseContentSpace ourSpace = BaseContentSpace(contentSpace);
-        //        require(msg.sender == owner || ourSpace.checkKMSAddr(msg.sender) > 0);
+        BaseContentSpace ourSpace = BaseContentSpace(contentSpace);
+        require(msg.sender == owner || ourSpace.checkKMSAddr(msg.sender) > 0);
 
         uint paramsLen = _opCodes.length;
 
-        require(_contentAddrs.length == paramsLen);
-        require(_requestNonces.length == paramsLen);
-        require(_userAddrs.length == paramsLen);
-        require(_ctxHashes.length == paramsLen);
-        require(_ts.length == paramsLen);
-
         for (uint i = 0; i < paramsLen; i++) {
```

Comment:
What is the background on these changes in 'require'?

Reply:
Just eliminating code that is not strictly necessary - because we're bumping up against size constraints.

It's quite justifiable to *not* validate these requirements because the execution will just revert if the arrays are not the same length. You could make the argument that this is more correct because it 'punishes' the caller for passing in bad data by consuming all gas.